tensorboard/plugins/debugger_v2/debug_data_multiplexer.py

```diff
@@ -98,6 +135,57 @@ def __init__(self, logdir):
         """
         self._logdir = logdir
         self._reader = None
+        self._reader_lock = threading.Lock()
+        self._createOrReloadReader()
+
+    def _createOrReloadReader(self):
+        """Creates or reloads reader for tfdbg2 data in the logdir.
+
+        If the reader has already been created, a new one will not be created;
+        instead, the existing reader's `update()` method will be unblocked on a
+        separate thread immediately.
+
+        If a reader has not been created, create it and start periodic calls to
+        `update()` on a separate thread.
+        """
+        with self._reader_lock:
+            if not self._reader:
+                try:
+                    from tensorflow.python.debug.lib import debug_events_reader
+                    from tensorflow.python.debug.lib import (
+                        debug_events_monitors,
+                    )
+
+                    self._reader = debug_events_reader.DebugDataReader(
+                        self._logdir
+                    )
+                    self._monitors = [
+                        debug_events_monitors.InfNanMonitor(
+                            self._reader, limit=DEFAULT_PER_TYPE_ALERT_LIMIT
+                        )
+                    ]
+                    # NOTE(cais): Currently each logdir is enforced to have only one
+                    # DebugEvent file set. So we add hard-coded default run name.
+                    run_repeatedly_in_background(self._reader.update)
+                    # TODO(cais): Start off a reading thread here, instead of being
+                    # called only once here.
+                except ImportError:
```

Comment:
I realize this is existing code, but especially when swallowing exceptions,  it's best practice to scope the `except` blocks at tightly as possible to avoid catching exceptions that we did not intend to catch.  E.g. catching a blanket `AttributeError` or `ValueError` could easily catch all kinds of exceptions other than the ones intended.

I'd strongly recommend scoping the errors to the minimum statement required (e.g. ImportError and AttributeError to _only_ the import statements, ValueError to _only_ the `DebugDataReader` construction) and would also recommend changing `DebugDataReader` so it raises a custom exception rather than a generic ValueError for a case like this where it's expected behavior that calling code must handle.

Reply:
Done tightening the try-except blocks in this method.